jwst/stpipe/step.py

```diff
@@ -661,8 +698,107 @@ def save_model(self, model, name, *args, **kwargs):
 
         # If the user specified an output_dir, replace the
         # original dirname with output_dir
-        if self.output_dir is not None:
-            dirname = self.output_dir
+        output_dir = self.search_attr('output_dir')
+        if output_dir is not None:
+            new_filename = join(output_dir, new_filename)
+        model.save(new_filename, *args, **kwargs)
+
+    @staticmethod
+    def make_output_path(step, data, basepath=None, suffix=None):
+        """Make up a path based on data and user specification
+
+        Parameters
+        ----------
+        step: Step
+            The step which produced the data
+
+        data: obj
+            Unused by this routine
```

Comment:
Might want to updated this docstring.  Looks like `data` is used for `meta.filename`.